### PR TITLE
chore: update exports and components for determined [WEB-1723]

### PR DESCRIPTION
**Pull request by johnkim-det**
- Add "Row" variation to `RadioGroup`
- Add `disabled` prop to `Toggle`
- Export additional types for `Checkbox`, `Select`, and `DatePicker`
- Update properties for `PivotProps`

supports https://github.com/determined-ai/determined/pull/8582

### CODE REVIEW

File: src/kit/RadioGroup.module.scss [hunk 1]
```diff
@@ -29,15 +29,47 @@
...
+          height: 32px;
+
+          &.label {
+            margin: 0;
+          }
+        }
+      }
+    }
+  }
+  .option > span > *:nth-child(2) {
```
Comment (EmilyBonar) [inline src/kit/RadioGroup.module.scss:72]: q: what is this selecting for?

File: src/kit/RadioGroup.module.scss [hunk 2]
```diff
@@ -29,15 +29,47 @@
...
-  .option {
-    padding: 0 var(--spacing-sm);
+  &.iconOnly {
+    .option {
+      padding: 0 var(--spacing-sm);
+    }
+  }
+  &.row {
+    display: flex;
+    gap: 8px;
```
Comment (EmilyBonar) [inline src/kit/RadioGroup.module.scss:42]: nit: use spacing

File: src/kit/RadioGroup.module.scss [hunk 3]
```diff
@@ -29,15 +29,47 @@
...
+    height: 80px;
+
+    > * {
+      flex-grow: 1;
+    }
+    .option {
+      border-color: var(--theme-surface-border);
+      border-inline-start: solid 1px var(--theme-surface-border);
+      border-radius: var(--theme-border-radius);
```
Comment (EmilyBonar) [inline src/kit/RadioGroup.module.scss:51]: q: should we just convert this to use Surface? seems like we're applying a lot of the same styles

File: src/kit/RadioGroup.module.scss [hunk 4]
```diff
@@ -29,15 +29,47 @@
...
+
+      &:hover {
+        border-color: var(--theme-status-active-weak);
+      }
+      &:not(:first-child)::before {
```
Comment (EmilyBonar) [inline src/kit/RadioGroup.module.scss:56]: q: what is this selecting for?

**Review by EmilyBonar — APPROVED** [review]
Looks good, just a few questions

File: src/kit/RadioGroup.module.scss (hunk 1 above)
Comment (johnkim-det) [reply src/kit/RadioGroup.module.scss:48]: was just preserving this from before, but not seeing a reason why it's there. will go ahead and remove it.

File: src/kit/RadioGroup.module.scss (hunk 3 above)
Comment (johnkim-det) [reply src/kit/RadioGroup.module.scss]: replacing `Radio.Button` with `Button` for "Row" variation and removing all of this CSS.

File: src/kit/RadioGroup.module.scss (hunk 4 above)
Comment (johnkim-det) [reply src/kit/RadioGroup.module.scss]: replacing `Radio.Button` with `Button` for "Row" variation and removing all of this CSS.

File: src/kit/RadioGroup.tsx [hunk 5]
```diff
@@ -18,7 +18,7 @@ interface Props {
   onChange?: (id: string | number) => void;
```
Comment (EmilyBonar) [inline src/kit/RadioGroup.tsx:18]: chore: after looking through the paired PR, i realized this interface could be made into a generic, something like `interface Props<T extends string | number>`. that way `defaultValue`, `onChange`, and `value` can all be constrained to the same type `T`

Comment (johnkim-det) [reply src/kit/RadioGroup.tsx:18]: good call, will do!

**Pull request merged**